src/vs/workbench/contrib/customEditor/common/customTextEditorModel.ts

```diff
@@ -58,12 +64,31 @@ export class CustomTextEditorModel extends Disposable implements ICustomEditorMo
 				this._onDidChangeContent.fire();
 			}
 		}));
+
+		this._register(extensionService.onWillStop(e => {
+			if (!this.isDirty()) {
+				return;
+			}
+
+			e.veto((async () => {
+				const didSave = await this.saveCustomEditor();
+				if (!didSave) {
+					// Veto
+					return true;
+				}
+				return false; // Don't veto
+			})(), localize('vetoExtHostRestart', "Custom editor '{0}' could not be saved.", this.name));
+		}));
```

Comment:
I would suggest to veto the extension host restart if it is automatically restarted, just like how it is done in notebooks land

https://github.com/microsoft/vscode/blob/e688e3a2c5331238ffa4cec490984fa427fda14e/src/vs/workbench/contrib/notebook/common/notebookEditorInput.ts#L88-L111

Reply:
If so, do we want to do that for the custom editor (non-text) as well?

Reply:
Yes

Reply:
Changed.

Reply:
I do not think it is aligned with the notebook editors implementation. Here is what it should do:

- If restart is automatic, veto always. 
- Otherwise, If not dirty, do not veto.
- Otherwise, Try to save and if save succeeds do not veto otherwise veto

Please follow the same for both custom editor models.

Reply:
Done